pkg/config/install_requirements.go

```diff
@@ -243,6 +243,16 @@ type IngressConfig struct {
 	DomainIssuerURL string `json:"domainIssuerURL,omitempty"`
 }
 
+// BuildPackConfig contains build pack info
+type BuildPackConfig struct {
+	// BuildPackName
+	BuildPackName string `json:"buildPackName,omitempty"`
```

Comment:
I am not a fan of `BuildPackName`. I think this is ambiguous and misleading (`jx edit buildpack` as well for that matter). A buildpack is the configuration used for building a given type of project. For example, when importing a Go project, you end up with a `jenkins-x.yaml` which looks like that:

```yaml
buildPack: go
```

See also https://jenkins-x.io/docs/reference/components/build-packs/.

When it comes to this `BuildPackConfig` we afaiu are not talking about configuring a specific buildpack, but rather change the lookup location for buildpacks in general, right? So we are more talking about a `BuildPackGroup` I would think. Or maybe `BuildPackLocation[Name|URL|Ref]`

I understand this is taking the `jx edit buildpack` concept 1-to-1 into requirements, but I don't think this is a good idea.

Maybe worth having a wider discussion around this.

Reply:
These keys are already being used to configure the dev environment in https://github.com/jenkins-x-charts/jxboot-resources/blob/master/jxboot-resources/values.yaml#L43-L45.

This is just making it configurable via `jx-requirements.yml`.

There's some disparity, yes. The `buildPackName` in `jxboot-resources` refers to the name of the **group** where the **buildPack** property in a given `jenkins-x.yml` file refers to a given pack within the group.

If we change this to `buildPackGroup`, we'd need to change the property in `jxboot-resources` too.

Reply:
@hferentschik I think what you suggested is something we should do in a follow up PR. But not this one as it's moving existing functionality to the `jx-requirements.yml` file.

Reply:
>  I think what you suggested is something we should do in a follow-up PR. But not this one as it's moving existing functionality to the jx-requirements.yml file.

but here is exactly the rub, we cannot just change the requirements at will. This is the interface/API to our users. We have already seen how hard it is to change things. We have deprecated fields, badly grouped fields, custom serialization to name just a few things.

Reply:
>  I think what you suggested is something we should do in a follow up PR. But not this one as it's moving existing functionality to the jx-requirements.yml file.

So just by adding these fields here, it is already configurable? How so? Are the values of the requirements file interpolated into the charts via the Helm templating?

Reply:
@mrageh at least we should have a wider discussion around this. What would it take to rename in the Helm chart? Can this be done without breaking things?

@dgozalo 
> If we change this to buildPackGroup, we'd need to change the property in jxboot-resources too.
Can this be done?

Reply:
There is imo another discussion we should have. I initially thought that if we add this information into `jx-requirements.yaml` we would the consume it somehow directly from there. What this seems to do is to somehow render this into the TeamSettings. We also add the requirements itself to the TeamSettings and then we look it up for some things and extract values from the stored requirements. At the same time, we also update the TeamSettings via interpolation into the jxboot-resources Chart. That is imo confusing, to say the least.

Reply:
@hferentschik let's hold of on this PR for now as you are right, the `jx-requirements.yml` is our public api and we should hold off on merging this.